dean.js

```diff
@@ -0,0 +1,136 @@
+/** @param {NS} ns **/
+/*
+create list of all servers
+iterate through list:
+count programs
+open ports
+nuke
+scp and exec hack script
+if all 5 ports are open then everything above has run, remove from list
+*/
+export async function main(ns) {
+	//create array of all servers except purchased called listServers[]
+	var listServers = await ns.scan('home');
+	let listIndex = 0;
+
+	while (listIndex < listServers.length) {
+		await ns.sleep(50);
+		let listScan = ns.scan(listServers[listIndex], true);
+		for (let i = 0; i < listScan.length; i++) {
+			if (listServers.indexOf(listScan[i]) === -1) {
+				listServers[listServers.length] = listScan[i];
```

Comment:
You're adding a new element to the array, you can do:

```suggestion
				listServers.push(listScan[i]);
```

Reply:
+